src/main/java/net/javadiscord/javabot/systems/qotw/submissions/SubmissionManager.java

```diff
@@ -91,68 +104,116 @@ public WebhookMessageCreateAction<?> handleSubmission(@NotNull ButtonInteraction
+		Responses.success(hook, "Submission Accepted",
+				"Successfully accepted submission by " + author.getAsMention()).queue();
+		notificationService.withQOTW(thread.getGuild()).sendSubmissionActionNotification(author, getOrRetrieveSubmission(thread), bestAnswer ? SubmissionStatus.ACCEPT_BEST : SubmissionStatus.ACCEPT);
+		Optional<ThreadChannel> newestPostOptional = config.getSubmissionsForumChannel().getThreadChannels()
+				.stream().max(Comparator.comparing(ThreadChannel::getTimeCreated));
+		if (newestPostOptional.isPresent()) {
+			ThreadChannel newestPost = newestPostOptional.get();
+			getMessagesByUser(thread, author).thenAccept(messages -> {
+				for (Message message : messages) {
+					if (message.getAuthor().isBot() || message.getType() != MessageType.DEFAULT) continue;
+					WebhookUtil.ensureWebhookExists(newestPost.getParentChannel().asForumChannel(), wh ->
+							WebhookUtil.mirrorMessageToWebhook(wh, message, message.getContentRaw(), newestPost.getIdLong()));
+				}
+				newestPost.sendMessageEmbeds(buildAuthorEmbed(author, bestAnswer)).queue();
+			});
 		}
+		thread.getManager().setLocked(true).setArchived(true).queue();
 	}
 
 	/**
-	 * Gets all active submission threads.
+	 * Declines a submission.
 	 *
-	 * @param guildId The ID of the guild to get the submission threads from
-	 * @return An immutable {@link List} of {@link QOTWSubmission}s.
+	 * @param hook   The {@link net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent} that was fired.
+	 * @param thread The submission's {@link ThreadChannel}.
+	 * @param author The submissions' author.
 	 */
-	public List<QOTWSubmission> getActiveSubmissionThreads(long guildId) {
-		try (Connection con = dbHelper.getDataSource().getConnection()) {
-			return qotwSubmissionRepository.getSubmissionsByQuestionNumber(guildId, qotwSubmissionRepository.getCurrentQuestionNumber());
-		} catch (SQLException e) {
-			ExceptionLogger.capture(e, getClass().getSimpleName());
-			return List.of();
-		}
+	public void declineSubmission(InteractionHook hook, @NotNull ThreadChannel thread, User author) {
+		thread.getManager().setName(SUBMISSION_DECLINED + thread.getName().substring(1)).queue();
+		notificationService.withQOTW(thread.getGuild(), author).sendSubmissionDeclinedEmbed();
+		Responses.success(hook, "Submission Declined", "Successfully declined submission by " + author.getAsMention()).queue();
+		notificationService.withQOTW(thread.getGuild()).sendSubmissionActionNotification(author, getOrRetrieveSubmission(thread), SubmissionStatus.DECLINE);
+		thread.getManager().setLocked(true).setArchived(true).queue();
+	}
+
+	private CompletableFuture<List<Message>> getMessagesByUser(@NotNull ThreadChannel channel, User user) {
+		return channel.getIterableHistory()
+				.reverse()
+				.takeAsync(channel.getMessageCount())
+				.thenApply(list -> list.stream().filter(m -> m.getAuthor().equals(user)).toList());
```

Comment:
Maybe system messages should also be allowed (and tested) due to things like the hug listener or message links.

Reply:
nah I feel like stuff like this should rather be excluded from submissions